engines/groovie/script.cpp

```diff
@@ -973,8 +973,13 @@ bool Script::playvideofromref(uint32 fileref, bool loopUntilAudioDone) {
 			// End the playback
 			return true;
 		}
-		_vm->_videoPlayer->fastForward();
-		_fastForwarding = true;
+		if (_fastForwarding == true) {
+			resetFastForward();
+			_vm->_videoPlayer->setOverrideSpeed(false);
+		} else {
+			_vm->_videoPlayer->fastForward();
+			_fastForwarding = true;
+		}
```

Comment:
This can be simplified as follows:

    if (_fastForwarding) {
        resetFastForward();
        _vm->_videoPlayer->setOverrideSpeed(false);
    } else {
        _vm->_videoPlayer->fastForward();
    }

    _fastForwarding = true;

Reply:
No. In case of reseting fast video playback _fastForwarding should be set to false, so it only should be set to true when we do fast forwarding video playback. Btw, in case of resetFastForward() won't set _fastForwarding to false (debug mode or smth) we should check it before setOverrideSpeed execution:

```
if (_fastForwarding) {
    resetFastForward();
    if (!_fastForwarding) 
        _vm->_videoPlayer->setOverrideSpeed(false);
} else {
    _fastForwarding = true;
    _vm->_videoPlayer->fastForward();
}
```

Reply:
Right. So it can be simplified like this then:

    if (!DebugMan.isDebugChannelEnabled(kDebugFast)) {
        _vm->_videoPlayer->setOverrideSpeed(false);
    } else {
        _vm->_videoPlayer->fastForward();
        _fastForwarding = true;
    }

Reply:
There is no sense to do comparison without setting _fastForwarding to false! In case of stop fast video playback _fastForwarding should be set to false and setOverrideSpeed(false) should be executed otherwise glithes will occur and goal won't be achieved (when you need to stop video fast playback you just click righ mouse once again and reset it).
I've already tested this in 7th guest. Works just fine.